### PR TITLE
N8N-3776-expressions-xss-part-2

### PR DESCRIPTION
**Pull request by alexgrozav**
Added window/global variable overwrite in eval context.

We will need to whitelist global variables/constructors that we want to expose in the application using `data`.

### CODE REVIEW

File: src/tmpl.js [hunk 1]
```diff
@@ -151,6 +151,11 @@ var tmpl = (function () {
 
     if (expr.slice(0, 11) !== 'try{return ') expr = 'return ' + expr
 
+    // Create local variable called window to prevent access to global window object
+    // var exposedVars = ['Date']
+    // expr = 'var window = { ' + exposedVars.map(function(varName) { return varName + ': ' + (typeof window !== 'object' ? 'global' : 'window') + '.' + varName }).join(', ') + ' }; ' + expr
```
Comment (ivov) [inline src/tmpl.js]: Leftover comments to remove or do they need to be preserved?

Comment (alexgrozav) [reply src/tmpl.js]: Good catch! Forgot about it.

File: test/specs/core.specs.js [hunk 2]
```diff
@@ -207,12 +207,15 @@ describe('riot-tmpl', function () {
...
+      expect(render('{ ' + (typeof window === 'object' ? 'window' : 'global') + ' }')).to.be.a('object')
+      expect(Object.keys(render('{ ' + (typeof window === 'object' ? 'window' : 'global') + ' }')).length).to.be(0)
+
+      expect(render('{' + (typeof window === 'object' ? 'window' : 'global') + '.globalVar }')).to.be(undefined)
       expect(render('{ this.$a }')).to.be(5)
       expect(render('{ void 0 }')).to.be(undefined)
       // without unprefixed global/window, default convertion to `new (D).Date()` throws here
-      data.Date = typeof window !== 'object' ? 'global' : 'window'
-      expect(render('{ new ' + data.Date + '.Date() }')).to.be.a('object')
+      data.Date = Date
```
Comment (ivov) [inline test/specs/core.specs.js:216]: Comment on L216 is outdated now? If so we should remove it.

Comment (alexgrozav) [reply test/specs/core.specs.js:216]: It is outdated. Removed!

**Review by ivov — APPROVED** [review]

**Pull request merged**